### PR TITLE
Install psycopg2 to the correct python virtual environment

### PR DESCRIPTION
**Pull request by tomwaldnz**
As per https://community.home-assistant.io/t/postgresql-error-during-connection-setup-no-module-named-psycopg2/72568/2

## Proposed change
Install psycopg2 to the homeassistant virtual environment

## Type of change

- [ ] Spelling, grammar or other readability improvements (`current` branch).
- [X] Adjusted missing or incorrect information in the current documentation (`current` branch).
- [ ] Added documentation for a new integration I'm adding to Home Assistant (`next` branch).
  - [ ] I've opened up a PR to add logo's and icons in [Brands repository](https://github.com/home-assistant/brands).
- [ ] Added documentation for a new feature I'm adding to Home Assistant (`next` branch).
- [ ] Removed stale or deprecated documentation.

## Additional information
See this thread https://community.home-assistant.io/t/postgresql-error-during-connection-setup-no-module-named-psycopg2/72568/2

- Link to parent pull request in the codebase: 
- Link to parent pull request in the Brands repository: 
- This PR fixes or closes issue: fixes #

## Checklist

- [ ] This PR uses the correct branch, based on one of the following:
  - I made a change to the existing documentation and used the `current` branch.
  - I made a change that is related to an upcoming version of Home Assistant and used the `next` branch.
- [X] The documentation follows the Home Assistant documentation [standards][].

[standards]: https://developers.home-assistant.io/docs/documenting/standards

### CODE REVIEW

**Comment by homeassistant** [comment]
Hi @tomwaldnz, 

It seems you haven't yet signed a CLA. Please do so [here](https://home-assistant.io/developers/cla_sign_start/?pr=home-assistant&#x2F;home-assistant.io#22613).

Once you do that we will be able to review and accept this pull request.

Thanks!

File: source/_integrations/recorder.markdown [hunk 1]
```diff
@@ -392,6 +392,18 @@ sudo apt-get install postgresql-server-dev-X.Y
...
+pip3 uninstall psycopg2
+sudo systemctl stop home-assistant@homeassistant.service
+sudo -u homeassistant -H -s
+source /srv/homeassistant/bin/activate
+pip3 install psycopg2
+```
+
+Then hit Control-D to exit the virtual environment.
+
+
```
Comment (emontnemery) [inline source/_integrations/recorder.markdown]: I think it's better to document that if Home Assistant is installed in a virtual environment, the additional libraries should be installed in the same virtual environment. This is also the case for mysql / mariadb.

**Review by emontnemery — CHANGES_REQUESTED** [review]

File: source/_integrations/recorder.markdown (hunk 1 above)
Comment (tomwaldnz) [reply source/_integrations/recorder.markdown]: I made changes based on your suggestion, but didn't notice that you had already made / suggested a change. I'm reasonably familiar with version control but not so much with HA or git. Apologies if I haven't done this quite correctly, and if you are able to fix it up please go ahead!

**Comment by tomwaldnz** [comment]
I've done my best to incorporate the suggestion from @emontnemery

File: source/_integrations/recorder.markdown [hunk 2]
```diff
@@ -371,6 +371,16 @@ sudo apt-get install default-libmysqlclient-dev libssl-dev
 pip3 install mysqlclient
 ```
 
+If Home Assistant is installed in a virtual environment, the additional libraries should be installed in the same virtual environment. If you installed Home Assistant Core to a Raspberry Pi following the [standard installation instructions](https://www.home-assistant.io/installation/raspberrypi) the following process should work.
```
Comment (emontnemery) [inline source/_integrations/recorder.markdown]: Let's rewrite this a little bit, the important point is to activate the venv before running pip3.
Also, the link should be relative.
```suggestion
If Home Assistant is installed in a virtual environment, the additional libraries should be installed in the same virtual environment. If you installed Home Assistant Core to a Raspberry Pi following the [standard installation instructions](/installation/raspberrypi) the venv can be activated like this prior to installing either `pymysql` or `mysqlclient`:
```

File: source/_integrations/recorder.markdown [hunk 3]
```diff
@@ -371,6 +371,16 @@ sudo apt-get install default-libmysqlclient-dev libssl-dev
...
+
+```bash
+pip3 uninstall psycopg2
+sudo systemctl stop home-assistant@homeassistant.service
+sudo -u homeassistant -H -s
+source /srv/homeassistant/bin/activate
+pip3 install mysqlclient
```
Comment (emontnemery) [inline source/_integrations/recorder.markdown]: Only the venv activation is relevant.
```suggestion
source /srv/homeassistant/bin/activate
pip3 install mysqlclient
deactivate
```

File: source/_integrations/recorder.markdown [hunk 4]
```diff
@@ -392,6 +402,19 @@ sudo apt-get install postgresql-server-dev-X.Y
 pip3 install psycopg2
 ```
 
+If Home Assistant is installed in a virtual environment, the additional libraries should be installed in the same virtual environment. If you installed Home Assistant Core to a Raspberry Pi following the [standard installation instructions](https://www.home-assistant.io/installation/raspberrypi) the following process should work.
```
Comment (emontnemery) [inline source/_integrations/recorder.markdown]: Let's rewrite this a little bit, the important point is to activate the venv before running pip3.
Also, the link should be relative.
```suggestion
If Home Assistant is installed in a virtual environment, the additional libraries should be installed in the same virtual environment. If you installed Home Assistant Core to a Raspberry Pi following the [standard installation instructions](/installation/raspberrypi) the venv can be activated like this prior to installing `psycopg2`:
```

File: source/_integrations/recorder.markdown [hunk 5]
```diff
@@ -392,6 +402,19 @@ sudo apt-get install postgresql-server-dev-X.Y
...
+
+```bash
+pip3 uninstall psycopg2
+sudo systemctl stop home-assistant@homeassistant.service
+sudo -u homeassistant -H -s
+source /srv/homeassistant/bin/activate
+pip3 install psycopg2
```
Comment (emontnemery) [inline source/_integrations/recorder.markdown]:
```suggestion
source /srv/homeassistant/bin/activate
pip3 install psycopg2
deactivate
```

File: source/_integrations/recorder.markdown [hunk 6]
```diff
@@ -392,6 +402,19 @@ sudo apt-get install postgresql-server-dev-X.Y
...
+```
+
+Then hit Control-D to exit the virtual environment.
+
+
```
Comment (emontnemery) [inline source/_integrations/recorder.markdown]:
```suggestion
```

**Comment by balloob** [comment]
This PR goes into too much detail. People that pick a virtualenv installation method are very advanced users and [are warned](https://www.home-assistant.io/installation/raspberrypi#install-home-assistant-core). We could add a collapsible hint for core users that they should not forget to activate their virtual environment. Instructions on how a virtual environment works or can be activated are outside of the scope of the recorder integration.

**Comment by tomwaldnz** [comment]
> This PR goes into too much detail. People that pick a virtualenv installation method are very advanced users and [are warned](https://www.home-assistant.io/installation/raspberrypi#install-home-assistant-core). We could add a collapsible hint for core users that they should not forget to activate their virtual environment. Instructions on how a virtual environment works or can be activated are outside of the scope of the recorder integration.

Agree it's advanced, but I'm a brand new Home Assistant user who used this method so I could install a couple of other things on my Raspberry Pi such as Pi Hole. So while I'm fairly technical, I'm also a beginner to HA. I can write a forum post or a blog post on my own website, but I thought having it here would be useful for others.

I will work out how to do an expandable section, hopefully that gets accepted.

**Review by tomwaldnz — COMMENTED** [review]
I think this change needs to be deleted as it's outdated now

**Comment by pvizeli** [comment]
We have community guides that help people to install HA Core. I'm sure your instruction are covered by someone there

**Comment by tomwaldnz** [comment]
> We have community guides that help people to install HA Core. I'm sure your instruction are covered by someone there

There is a guide for install, but there's no guide I can find for moving over to a standalone database. I could write a whole guide, but I figured adding three lines to this page about using standalone databases would make more sense, to centralize the information.

**Pull request closed without merging**